### PR TITLE
[BugFix] Support audit for insert into statement executing from follower

### PR DESCRIPTION
**Pull request by liuyehcf**
This is an continuation work of #29901, and it cannot work well when there are multi frontend instances.

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 2 / 32 (06.25%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|0|27|00.00%|[887, 888, 889, 890, 891, 892, 893, 894, 897, 898, 899, 900, 901, 902, 903, 904, 905, 907, 908, 909, 910, 912, 913, 914, 915, 916, 917]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|4|25.00%|[483, 484, 1376]|
|:large_blue_circle:|com/starrocks/sql/ast/InsertStmt.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 35 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.h|0|3|00.00%|[110, 112, 121]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.cpp|0|3|00.00%|[195, 196, 225]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.h|0|5|00.00%|[44, 45, 89, 91, 99]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|2|00.00%|[64, 65]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.h|0|1|00.00%|[106]|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.cpp|0|1|00.00%|[67]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.h|0|6|00.00%|[39, 40, 82, 83, 84, 92]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.cpp|0|5|00.00%|[141, 142, 200, 203, 206]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.h|0|7|00.00%|[51, 108, 109, 110, 111, 112, 120]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.cpp|0|2|00.00%|[49, 50]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 2 / 32 (06.25%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|0|27|00.00%|[887, 888, 889, 890, 891, 892, 893, 894, 897, 898, 899, 900, 901, 902, 903, 904, 905, 907, 908, 909, 910, 912, 913, 914, 915, 916, 917]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|4|25.00%|[483, 484, 1376]|
|:large_blue_circle:|com/starrocks/sql/ast/InsertStmt.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 35 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.h|0|3|00.00%|[110, 112, 121]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.cpp|0|3|00.00%|[195, 196, 225]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.h|0|5|00.00%|[44, 45, 89, 91, 99]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|2|00.00%|[64, 65]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.h|0|1|00.00%|[106]|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.cpp|0|1|00.00%|[67]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.h|0|6|00.00%|[39, 40, 82, 83, 84, 92]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.cpp|0|5|00.00%|[141, 142, 200, 203, 206]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.h|0|7|00.00%|[51, 108, 109, 110, 111, 112, 120]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.cpp|0|2|00.00%|[49, 50]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 17 / 32 (53.12%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|4|25.00%|[483, 484, 1376]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|15|27|55.56%|[904, 905, 907, 908, 909, 910, 912, 913, 914, 915, 916, 917]|
|:large_blue_circle:|com/starrocks/sql/ast/InsertStmt.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 35 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.h|0|3|00.00%|[110, 112, 121]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.cpp|0|3|00.00%|[195, 196, 225]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.h|0|5|00.00%|[44, 45, 89, 91, 99]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|2|00.00%|[64, 65]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.h|0|1|00.00%|[106]|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.cpp|0|1|00.00%|[67]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.h|0|6|00.00%|[39, 40, 82, 83, 84, 92]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.cpp|0|5|00.00%|[141, 142, 200, 203, 206]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.h|0|7|00.00%|[51, 108, 109, 110, 111, 112, 120]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.cpp|0|2|00.00%|[49, 50]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 17 / 32 (53.12%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|4|25.00%|[483, 484, 1376]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|15|27|55.56%|[904, 905, 907, 908, 909, 910, 912, 913, 914, 915, 916, 917]|
|:large_blue_circle:|com/starrocks/sql/ast/InsertStmt.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 35 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.h|0|3|00.00%|[110, 112, 121]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.cpp|0|3|00.00%|[195, 196, 225]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.h|0|5|00.00%|[44, 45, 89, 91, 99]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|2|00.00%|[64, 65]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.h|0|1|00.00%|[106]|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.cpp|0|1|00.00%|[67]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.h|0|6|00.00%|[39, 40, 82, 83, 84, 92]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.cpp|0|5|00.00%|[141, 142, 200, 203, 206]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.h|0|7|00.00%|[51, 108, 109, 110, 111, 112, 120]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.cpp|0|2|00.00%|[49, 50]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 17 / 32 (53.12%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|4|25.00%|[483, 484, 1376]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|15|27|55.56%|[904, 905, 907, 908, 909, 910, 912, 913, 914, 915, 916, 917]|
|:large_blue_circle:|com/starrocks/sql/ast/InsertStmt.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 35 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.h|0|3|00.00%|[110, 112, 121]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.cpp|0|3|00.00%|[195, 196, 225]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.h|0|5|00.00%|[44, 45, 89, 91, 99]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|2|00.00%|[64, 65]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.h|0|1|00.00%|[106]|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.cpp|0|1|00.00%|[67]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.h|0|6|00.00%|[39, 40, 82, 83, 84, 92]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.cpp|0|5|00.00%|[141, 142, 200, 203, 206]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.h|0|7|00.00%|[51, 108, 109, 110, 111, 112, 120]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.cpp|0|2|00.00%|[49, 50]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 17 / 32 (53.12%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|4|25.00%|[483, 484, 1376]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|15|27|55.56%|[904, 905, 907, 908, 909, 910, 912, 913, 914, 915, 916, 917]|
|:large_blue_circle:|com/starrocks/sql/ast/InsertStmt.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 35 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.h|0|3|00.00%|[110, 112, 121]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.cpp|0|3|00.00%|[195, 196, 225]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.h|0|5|00.00%|[44, 45, 89, 91, 99]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|2|00.00%|[64, 65]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.h|0|1|00.00%|[106]|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.cpp|0|1|00.00%|[67]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.h|0|6|00.00%|[39, 40, 82, 83, 84, 92]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.cpp|0|5|00.00%|[141, 142, 200, 203, 206]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.h|0|7|00.00%|[51, 108, 109, 110, 111, 112, 120]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.cpp|0|2|00.00%|[49, 50]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 17 / 32 (53.12%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|4|25.00%|[483, 484, 1376]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|15|27|55.56%|[904, 905, 907, 908, 909, 910, 912, 913, 914, 915, 916, 917]|
|:large_blue_circle:|com/starrocks/sql/ast/InsertStmt.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 35 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.h|0|3|00.00%|[110, 112, 121]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.cpp|0|3|00.00%|[195, 196, 225]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.h|0|5|00.00%|[44, 45, 89, 91, 99]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|2|00.00%|[64, 65]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.h|0|1|00.00%|[106]|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.cpp|0|1|00.00%|[67]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.h|0|6|00.00%|[39, 40, 82, 83, 84, 92]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.cpp|0|5|00.00%|[141, 142, 200, 203, 206]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.h|0|7|00.00%|[51, 108, 109, 110, 111, 112, 120]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.cpp|0|2|00.00%|[49, 50]|

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 1]
```diff
@@ -480,6 +480,10 @@ public void execute() throws Exception {
                 // Write the resource group information to audit log.
                 prepareResourceGroup(context, ResourceGroupClassifier.QueryType.fromStatement(parsedStmt));
 
+                Preconditions.checkState(!(parsedStmt instanceof InsertStmt && parsedStmt.isExplain()
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:483]: why not forward to leader FE?

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 2]
```diff
@@ -1368,6 +1372,9 @@ private void handleShow() throws IOException, AnalysisException, DdlException {
     }
 
     private void handleExplainStmt(String explainString) throws IOException {
+        if (explainString == null) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:1375]: Why explainString could be null?

File: fe/fe-core/src/main/java/com/starrocks/qe/DefaultCoordinator.java [hunk 3]
```diff
@@ -882,22 +883,39 @@ public void updateFragmentExecStatus(TReportExecStatusParams params) {
...
-    public void updateAuditStatistics(TReportAuditStatisticsParams params) {
-        auditStatistics = new PQueryStatistics();
-        auditStatistics.scanRows = params.scan_rows;
-        auditStatistics.scanBytes = params.scan_bytes;
-        auditStatistics.returnedRows = params.returned_rows;
-        auditStatistics.cpuCostNs = params.cpu_cost_ns;
-        auditStatistics.memCostBytes = params.mem_cost_bytes;
-        auditStatistics.spillBytes = params.spill_bytes;
-        if (CollectionUtils.isNotEmpty(params.stats_items)) {
+    public synchronized void updateAuditStatistics(TReportAuditStatisticsParams params) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/DefaultCoordinator.java:886]: data sink is in the one single node, and you already have this check  `    if (_num_sinkers.fetch_sub(1, std::memory_order_acq_rel) == 1) {`, why this method will be concurrent?

Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/qe/DefaultCoordinator.java:886]: I'm not sure about that, maybe there are many backend will call this function concurrently.

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java (hunk 2 above)
Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java]: Reverted

**Review by ZiheLiu — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 38 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.h|0|3|00.00%|[110, 112, 121]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.cpp|0|3|00.00%|[195, 196, 225]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.h|0|5|00.00%|[44, 45, 89, 91, 99]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|2|00.00%|[64, 65]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.h|0|1|00.00%|[106]|
|:large_blue_circle:|src/exec/pipeline/pipeline_driver_executor.cpp|0|2|00.00%|[348, 349]|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.cpp|0|1|00.00%|[67]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.h|0|6|00.00%|[39, 40, 82, 83, 84, 92]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.cpp|0|5|00.00%|[141, 142, 200, 203, 206]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.h|0|7|00.00%|[51, 108, 109, 110, 111, 112, 120]|
|:large_blue_circle:|src/runtime/query_statistics.cpp|0|1|00.00%|[58]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.cpp|0|2|00.00%|[49, 50]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 26 / 67 (38.81%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|0|31|00.00%|[888, 889, 890, 892, 893, 894, 895, 896, 897, 898, 899, 902, 903, 904, 905, 906, 907, 908, 909, 910, 911, 912, 913, 914, 915, 917, 918, 919, 920, 921, 922]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|14|24|58.33%|[1388, 1389, 1390, 1507, 1508, 1509, 1510, 1511, 1512, 1513]|
|:large_blue_circle:|com/starrocks/qe/LeaderOpExecutor.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/InsertStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 38 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.h|0|3|00.00%|[110, 112, 121]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.cpp|0|3|00.00%|[195, 196, 225]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.h|0|5|00.00%|[44, 45, 89, 91, 99]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|2|00.00%|[64, 65]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.h|0|1|00.00%|[106]|
|:large_blue_circle:|src/exec/pipeline/pipeline_driver_executor.cpp|0|2|00.00%|[348, 349]|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.cpp|0|1|00.00%|[67]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.h|0|6|00.00%|[39, 40, 82, 83, 84, 92]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.cpp|0|5|00.00%|[141, 142, 200, 203, 206]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.h|0|7|00.00%|[51, 108, 109, 110, 111, 112, 120]|
|:large_blue_circle:|src/runtime/query_statistics.cpp|0|1|00.00%|[58]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.cpp|0|2|00.00%|[49, 50]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 62 / 117 (52.99%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/AuditStatisticsUtil.java|46|95|48.42%|[27, 30, 40, 41, 42, 43, 44, 45, 46, 47, 54, 58, 64, 70, 76, 82, 88, 93, 94, 96, 97, 98, 99, 100, 102, 103, 104, 105, 106, 108, 110, 111, 112, 114, 117, 118, 120, 126, 136, 148, 149, 150, 151, 153, 154, 156, 157, 159, 160]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|5|10|50.00%|[1409, 1410, 1411, 1544, 1545]|
|:large_blue_circle:|com/starrocks/qe/LeaderOpExecutor.java|3|4|75.00%|[122]|
|:large_blue_circle:|com/starrocks/sql/ast/InsertStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 38 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.h|0|3|00.00%|[114, 117, 125]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.cpp|0|3|00.00%|[195, 196, 225]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.h|0|5|00.00%|[44, 45, 89, 91, 99]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|2|00.00%|[64, 65]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.h|0|1|00.00%|[106]|
|:large_blue_circle:|src/exec/pipeline/pipeline_driver_executor.cpp|0|2|00.00%|[348, 349]|
|:large_blue_circle:|src/exec/pipeline/result_sink_operator.cpp|0|1|00.00%|[68]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.h|0|6|00.00%|[39, 40, 82, 83, 84, 92]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.cpp|0|5|00.00%|[141, 142, 200, 203, 206]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.h|0|7|00.00%|[51, 108, 109, 110, 111, 112, 120]|
|:large_blue_circle:|src/runtime/query_statistics.cpp|0|1|00.00%|[58]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.cpp|0|2|00.00%|[49, 50]|

**Review by ZiheLiu — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/InsertStmt.java [hunk 4]
```diff
@@ -218,7 +218,7 @@ public void setPartitionNotSpecifiedInOverwrite(boolean partitionNotSpecifiedInO
 
     @Override
     public RedirectStatus getRedirectStatus() {
-        if (isExplain()) {
+        if (isExplain() && !StatementBase.ExplainLevel.ANALYZE.equals(getExplainLevel())) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/InsertStmt.java:221]: why need this check?

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by liuyehcf** [comment]
@Mergifyio backport branch-3.2

**Pull request merged**